### PR TITLE
Makefile: Create GPG signatures for tarballs and git tags

### LINKED ISSUE #625: GPG signed releases
ESS should probably provide GPG signatures for releases so that people installing from the tarball can be sure the tarball they received is the one we actually released. Given that many of our users are high-value targets (people with access to all sorts of data --- health info, financial information, etc), it would be a *bad thing* if attackers were to modify ESS since it can run arbitrary code.

As far as I know, there are no downsides to providing GPG signatures with release tarballs.

### PR DESCRIPTION
**Pull request by jabranham**
Closes #625

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -122,7 +126,8 @@ ChangeLog: VERSION
 tag:
 	@echo "** Tagging the release **  1) pull existing;  2) tag  3) push it"
 	git pull --tags
-	git tag -a -m'release tagging' v$(ESSVERSION)
+	@echo "Creating tag and signing using gpg"
+	git tag -s -m'release tagging' v$(ESSVERSION)
```
Comment (vspinu) [inline Makefile]: Don't we have to push the tags explicitly or the `git push` from below does the job?

Comment (jabranham) [reply Makefile]: I believe the git push does that.

**Comment by jabranham** [comment]
@vspinu OK to merge?

**Comment by mmaechler** [comment]
Thank you @jabranham .  I've switched to this branch and tried  `make tarballs`  which ended with an error,  I can also reproduce directly :

       gpg -ba -o ess-18.10.tgz.sig ess-18.10.tgz

which gives
```
gpg: no default secret key: secret key not available
gpg: signing failed: secret key not available
```

**Comment by jabranham** [comment]
Sounds like you don't have gpg set up on that computer correctly. I'm assuming `gpg --list-secret-keys` doesn't return anything?

You can teach gpg about a new secret key with the `--import` option.

**Comment by mmaechler** [comment]
No, there *is* exactly one key listed (the newly created one with 4096 bit) by  `gpg --list-secret-keys`.
But there are many more listed if I list  --list-keys ...
I see I can add    `-u maechler@ethz.ch` to the argument list and then it seems it *would* work
if I remembered the my secret key ;-\(   -- I'll look later today on my home computer I may have "remembered" the key there...

**Comment by mmaechler** [comment]
Yes, of course, --list-keys contains many, but also the one from `--list-secret-keys`

Fedora Linux 28 is quite recent, and the system very well supported by our IT. I've decided a few years ago to go with that and devote my time to other things than OS + system tools maintenance.
Hence I use its gpg, a version 1.4.23 and really don't want to manually maintain a newer version..

**Comment by mmaechler** [comment]
The current master `Makefile` works with current `Makeconf`  and so this branch becomes obsolete.
After all the  comments thread was the important thing here...

**Pull request closed without merging**